src/raftstore/coprocessor/split_check/rows.rs

```diff
@@ -0,0 +1,266 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use raftstore::store::{util, Msg};
+use rocksdb::DB;
+use util::transport::{RetryableSendCh, Sender};
+
+use super::super::metrics::*;
+use super::super::{Coprocessor, KeyEntry, ObserverContext, SplitCheckObserver, SplitChecker};
+use super::Host;
+
+pub struct Checker {
+    max_rows: u64,
+    split_rows: u64,
+    current_rows: u64,
+    split_key: Option<Vec<u8>>,
+}
+
+impl Checker {
+    pub fn new(max_rows: u64, split_rows: u64) -> Checker {
+        Checker {
+            max_rows,
+            split_rows,
+            current_rows: 0,
+            split_key: None,
+        }
+    }
+}
+
+impl SplitChecker for Checker {
+    fn on_kv(&mut self, _: &mut ObserverContext, row: &KeyEntry) -> bool {
```

Comment:
Actually, the `row` here is just one version of that row.
If you want to calculate the number of rows, you need to check if the current key is in the same row as the previous key.

Reply:
I think we could simply consider each key has only one version here and pay more attention to the versions in GC?

Reply:
ping @huachaohuang